### ISSUE TITLE
Hash instances derived from Hash[Long] (e.g. Hash[List[Long]]) do not pass laws

### ISSUE THREAD

**Issue by joroKr21** [typelevel/cats#2878]
The failing law is `sameAsUniversalHash`:
```scala
List(-1L).hashCode // -1698283412
Hash[List[Long]].hash(List(-1L)) // 1918838495
-1L.hashCode // 0
-1L.## // -1
```

Took me a while to realize this while working on `Hash` derivation over at `kittens`.

To be honest I'm not sure what's the point of that law.
Why would `Hash` exist if it's bound to be the same as `hashCode`?

**Comment by ctongfei** [comment]
I'm the author who wrote that law.

The law `sameAsUniversalHash` is supposed to guarantee for common types and case classes, the derived `Hash` instances behave in the same way as `.hashCode` (Not for user-defined `Hash` instance s).

The standard library's `.hashCode` implementation for `List`s can be found at https://github.com/scala/scala/blob/2.12.x/src/library/scala/util/hashing/MurmurHash3.scala#L110. Note that they use `.##` instead of `.hashCode`. In the original PR for `Hash` (#1712), the community decided to use `.hashCode` instead of `.##` for performance reasons. So I would say that this behavior is expected.

**Comment by joroKr21** [comment]
Makes sense to me :+1:

**Comment by ctongfei** [comment]
@joroKr21 So for your work in `kittens`, you should just test the `hashCompatibility` property.

**Comment by joroKr21** [comment]
This should be addressed by #4139
